src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,43 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private int[] array;
+    private int n;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+        this.n = array.length;
+    }
+
+    @Override
+    protected void compute() {
+        if (array.length == 1) {
+            return;
+        }
+        var left = Arrays.copyOfRange(array, 0 , n/2);
+        var right = Arrays.copyOfRange(array, n/2 , n);
+        var leftTask = new MergeSortAction(left);
+        var rightTask = new MergeSortAction(right);
+        leftTask.fork();
```

Comment:
Consider using 'invokeAll' to simplify the forking of tasks and waiting for their completion. It can replace the manual fork, compute, and join calls.